egcg_core/rest_communication.py

```diff
@@ -20,6 +22,58 @@ def serialise(queries):
             serialised_queries[k] = json.dumps(v) if type(v) is dict else v
         return serialised_queries
 
+    @staticmethod
+    def get_file_content(file_name):
+        """
+        This method open in binary mode th file provided.
+        :param file_name: path to the file
+        :return: file content
+        """
+        with open(file_name, 'rb') as open_f:
+            fc = open_f.read()
+        return fc
+
+    @staticmethod
+    def _extract_files(json_dict):
+        """
+        This method take a json dict and will search all the values for a tuple where the first element is "file".
+        It extract the value, read the content of the file and create another tuple ready to be used by requests
+        This method will modify the original json_dict and remove the file entry from it.
+        :param json_dict: The json dictionary to search
+        :return: a dict where the keys are the field to update and the value are tuple with (filename, file content, file MIME type)
+        """
+        files = {}
+        for k in json_dict:
+            if isinstance(json_dict[k], tuple) and len(json_dict[k]) > 1 and json_dict[k][0] == 'file':
+                # Construct the 3 parts file upload tuple as requests requires it
+                files[k] = (
+                    json_dict[k][1],                                 # file name
+                    Communicator.get_file_content(json_dict[k][1]),  # file content
+                    mimetypes.guess_type(json_dict[k][1])[0]         # file mime type
+                )
+        if files:
+            # remove the extracted files to the original json
+            for k in files:
+                json_dict.pop(k)
+            return files
+        return None
```

Comment:
If we return `None` here, could we end up trying to push `{'fastqc_report': null}` to the Rest API? If so, we should either make that nullable or return an empty list here instead.